itests/hive-unit/src/test/java/org/apache/hive/jdbc/TestJdbcDriver2.java

```diff
@@ -2793,6 +2795,124 @@ public void run() {
+    stmt.execute("create database " + primaryDb + " with dbproperties('repl.source.for'='1,2,3')");
+    stmt.execute("create table " + primaryTblName + " (id int)");
+    stmt.execute("insert into " + primaryTblName + " values (1), (2)");
+    stmt.close();
+
+    // Test query logs for bootstrap dump and load
+    String[] expectedBootstrapDumpLogs = {
+            "REPL::START",
+            "REPL::TABLE_DUMP",
+            "REPL::END"
+    };
+
+    // Bootstrap dump
+    stmt = (HiveStatement)con.createStatement();
+    advanceDumpDir();
+    ResultSet replDumpRslt = stmt.executeQuery("repl dump " + primaryDb +
+            " with ('hive.repl.rootdir' = '" + replDir + "')");
+    assertTrue(replDumpRslt.next());
+    String dumpLocation = replDumpRslt.getString(1);
+    String lastReplId = replDumpRslt.getString(2);
+    List<String> logs = stmt.getQueryLog(false, 10000);
+    stmt.close();
+    LOG.error("Query_Log for Bootstrap Dump");
+    verifyFetchedLog(logs, expectedBootstrapDumpLogs);
+
+    String[] expectedBootstrapLoadLogs = {
+            "REPL::START",
+            "REPL::TABLE_LOAD",
+            "REPL::END"
+    };
+
+    // Bootstrap load
+    stmt = (HiveStatement)con.createStatement();
+    stmt.execute("repl load " + replicaDb + " from '" + dumpLocation + "'");
+    logs = stmt.getQueryLog(false, 10000);
+    stmt.close();
+    LOG.error("Query_Log for Bootstrap Load");
+    verifyFetchedLog(logs, expectedBootstrapLoadLogs);
+
+    // Perform operation for incremental replication
+    stmt = (HiveStatement)con.createStatement();
+    stmt.execute("insert into " + primaryTblName + " values (3), (4)");
+    stmt.close();
+
+    // Test query logs for incremental dump and load
+    String[] expectedIncrementalDumpLogs = {
+            "REPL::START",
+            "REPL::EVENT_DUMP",
+            "REPL::END"
+    };
+
+    // Incremental dump
+    stmt = (HiveStatement)con.createStatement();
+    advanceDumpDir();
+    replDumpRslt = stmt.executeQuery("repl dump " + primaryDb + " from " + lastReplId +
+            " with ('hive.repl.rootdir' = '" +  replDir + "')");
+    assertTrue(replDumpRslt.next());
+    dumpLocation = replDumpRslt.getString(1);
+    lastReplId = replDumpRslt.getString(2);
+    logs = stmt.getQueryLog(false, 10000);
+    stmt.close();
+    LOG.error("Query_Log for Incremental Dump");
+    verifyFetchedLog(logs, expectedIncrementalDumpLogs);
+
+    String[] expectedIncrementalLoadLogs = {
+            "REPL::START",
+            "REPL::EVENT_LOAD",
+            "REPL::END"
+    };
+
+    // Incremental load
+    stmt = (HiveStatement)con.createStatement();
+    stmt.execute("repl load " + replicaDb + " from '" + dumpLocation + "'");
+    logs = stmt.getQueryLog(false, 10000);
+    stmt.close();
+    LOG.error("Query_Log for Incremental Load");
+    verifyFetchedLog(logs, expectedIncrementalLoadLogs);
+
+    // DB cleanup
+    stmt = (HiveStatement)con.createStatement();
```

Comment:
Do we need to catch any exception from previous statements and do cleanup in finally block ? or is it best to do it in tearDownAfterClass?

Reply:
done in finally block as many settings are local to this test